client/verta/verta/deployment/_deployedmodel.py

```diff
@@ -233,99 +240,109 @@ def predict(
-                except ValueError:  # not JSON response; 502 not from model back end
-                    pass
-                else:  # from model back end; contains message (maybe)
-                    # try to directly print message, otherwise line breaks appear as '\n'
-                    msg = data.get("message") or json.dumps(data)
-                    raise RuntimeError(
-                        "deployed model encountered an error: {}".format(msg)
-                    )
-            elif not (
-                response.status_code >= 500 or response.status_code in (404, 429)
-            ):  # clientside error
-                break
-
-            sleep = 0.3 * (2 ** (num_retries + 1))
-            print(
-                "received status {}; retrying in {:.1f}s".format(
-                    response.status_code, sleep
-                )
+            self,
+            x: List[Any],
+            compress=False,
+            max_retries: int = http_session.DEFAULT_MAX_RETRIES,
+            retry_status: Set[int] = http_session.DEFAULT_STATUS_FORCELIST,
+            backoff_factor: float = http_session.DEFAULT_BACKOFF_FACTOR,
+            prediction_id: str = None,
+            ) -> Tuple[str, Dict[str, Any]]:
+        """
+        Makes a prediction using input `x` the same as `predict`, but returns a tuple including the ID of the
+        prediction request along with the prediction results.
+
+        .. versionadded:: 0.22.0
+           The `prediction_id` parameter.
+        .. versionadded:: 0.22.0
+           The `retry_status` parameter.
+        .. versionadded:: 0.22.0
+           The `backoff_factor` parameter.
+
+        Parameters
+        ----------
+        x : list
+            A batch of inputs for the model.
+        compress : bool, default False
+            Whether to compress the request body.
+        max_retries : int, default 13
+            Maximum number of retries on status codes listed in ``retry_status`` parameter only.
+        retry_status : set, default {404, 429}
+            Set of status codes, as integers, for which retry attempts should be made.  Overwrites default value of
+            {404, 429}.  Expand the set to include more: ``retry_status={404, 429, <NEW_STATUS_CODE>}``
+        backoff_factor : float, default 0.3
+            A backoff factor to apply between retry attempts.  Uses standard urllib3 sleep pattern:
+            ``{backoff factor} * (2 ** ({number of total retries} - 1))`` with a maximum sleep time between requests of
+            120 seconds.
+        prediction_id: str, default None
+            A custom str to use as the ID for the prediction request.  Defaults to a randomly generated numeric id.
+        Returns
+        -------
+        id, prediction: Tuple(str, list)
+            The first element of the tuple is the prediction ID.
+            The second element of the tuple is the output returned by the deployed model for `x`.
```

Comment:
fyi, to be faithful to numpydoc conventions, this would be [more like](https://numpydoc.readthedocs.io/en/stable/format.html#returns)

```suggestion
        id : str
            The prediction ID.
        prediction: List[Any]
            The output returned by the deployed model for `x`.
```

(since it's arguably self-explanatory that a function with "multiple return values" returns a `tuple`)

though I'm not personally sure how that interacts with static type analysis tooling.